django/template/loaders/cached.py

```diff
@@ -26,12 +26,25 @@ def __init__(self, engine, loaders):
     def get_contents(self, origin):
         return origin.loader.get_contents(origin)
 
+    def _cache_exception(self, exc):
+        if isinstance(exc, type) and issubclass(exc, TemplateDoesNotExist):
+            return exc
+        elif isinstance(exc, TemplateDoesNotExist):
+            if self.engine.debug:
+                return exc.__class__(*exc.args, tried=exc.tried)
+            return exc.__class__
```

Comment:
Just having a look at the test failure there, looks like I'll need to change the approach a little.

We still want to cache the raw request, but `TemplateDoesNotExist` has a required positional argument, so can't be raised "raw".